modules/kubernetes/grafana-agent/charts/grafana-agent-extras/templates/monitors.yaml

```diff
@@ -19,3 +19,94 @@ spec:
       app.kubernetes.io/name: ingress-nginx
       function: metrics
 {{ end }}
+{{ if .Values.includeKubeletMetrics }}
+---
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  labels:
+    xkf.xenit.io/monitoring: tenant
+  name: kubelet-monitor
+  namespace: grafana-agent
+spec:
+  endpoints:
+    - bearerTokenFile: /var/run/secrets/kubernetes.io/serviceaccount/token
+      honorLabels: true
+      metricRelabelings:
+        - action: keep
+          regex: {{ .Values.kubeletMetricsNamespaces }}
+          sourceLabels:
+            - namespace
+        - action: keep
+          regex: container_cpu_usage_seconds_total|container_network_transmit_bytes_total|container_network_receive_bytes_total|container_memory_swap|container_network_receive_packets_total|container_cpu_cfs_periods_total|container_cpu_cfs_throttled_periods_total|node_namespace_pod_container:container_cpu_usage_seconds_total:sum_rate|container_memory_working_set_bytes|container_memory_rss|container_network_receive_packets_dropped_total|container_start_time_seconds|container_network_transmit_packets_dropped_total|container_memory_cache|container_network_transmit_packets_total
+          sourceLabels:
+            - __name__
+        - action: labeldrop
+          regex: endpoint|id|job|metrics_path|service|name|instance
```

Comment:
Don't know how the final output looks. 
Can't remember what service says but can it be useful?

Reply:
Not really, it is always set to "kubelet".